boards/hail/src/io.rs

```diff
@@ -11,6 +11,20 @@ pub static mut WRITER: Writer = Writer { initialized: false };
 
 impl Write for Writer {
     fn write_str(&mut self, s: &str) -> ::core::fmt::Result {
+        // XXX Replace with something like kernel::begin_panic()
+        // XXX Maybe place that call at panic_fmt, as it's called first
+        // XXX Better to cancel the transaction rather than hope we wait long enough
+        // Let any outstanding uart DMA's finish
+        unsafe {
+            asm!("nop");
+            asm!("nop");
+            for _ in 0..200000 {
+                asm!("nop");
+            }
+            asm!("nop");
+            asm!("nop");
```

Comment:
Moreover, these should be moved out of `write_str` and into the top of `panic_fmt`